File: Dev/unitTest_Engine_cs/Graphics/Transition/Transition.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace unitTest_Engine_cs.Graphics.Transition
{
	class Transition : EngineTest
	{
		public Transition()
			: base(150)
		{
		}

		protected override void OnStart()
		{
		}

		protected override void OnUpdating()
		{
			if(Time == 9)
			{
				var texture = asd.Engine.Graphics.CreateTexture2D("Data/Texture/Sample1.png");

				var obj = new asd.TextureObject2D();
				obj.Texture = texture;
				var scene = new asd.Scene();
				var layer = new asd.Layer2D();
				layer.AddObject(obj);
				scene.AddLayer(layer);

				var transition = new asd.TransitionFade(60.0f / 60.0f, 60.0f / 60.0f);
				asd.Engine.ChangeSceneWithTransition(scene, transition);
			}
		}
	}
}

File: Dev/unitTest_Engine_cpp_gtest/Graphics/Transition/Transition.cpp

#include <gtest/gtest.h>
#include <Altseed.h>
#include "../../EngineTest.h"

class Graphics_Transition : public EngineTest
{
public:
	Graphics_Transition(bool isOpenGLMode)
		: EngineTest(asd::ToAString("Transition"), isOpenGLMode, 150)
	{
	}

private:

protected:
	void OnStart()
	{
	}

	void OnUpdating()
	{
		if (GetTime() == 9)
		{
			auto texture = asd::Engine::GetGraphics()->CreateTexture2D(asd::ToAString("Data/Texture/Sample1.png").c_str());

			auto obj = std::make_shared<asd::TextureObject2D>();
			obj->SetTexture(texture);
			auto scene = std::make_shared<asd::Scene>();
			auto layer = std::make_shared<asd::Layer2D>();
			layer->AddObject(obj);
			scene->AddLayer(layer);

			auto transition = std::make_shared<asd::TransitionFade>(60.0f/60.0f, 60.0f/60.0f);
			asd::Engine::ChangeSceneWithTransition(scene, transition);
		}
	}
};


ENGINE_TEST(Graphics, Transition)
